lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -54,27 +59,48 @@ public TaskServiceImpl(TaskanaEngine taskanaEngine, TaskMapper taskMapper,
         this.taskanaEngineImpl = (TaskanaEngineImpl) taskanaEngine;
         this.taskMapper = taskMapper;
         this.objectReferenceMapper = objectReferenceMapper;
+        this.workbasketService =  taskanaEngineImpl.getWorkbasketService();
     }
 
     @Override
-    public Task claim(String id, String userName) throws TaskNotFoundException {
+    public Task claim(String id)  throws TaskNotFoundException, InvalidStateException, InvalidOwnerException {
+        return claim(id, false);
+    }
+
+    @Override
+    public Task claim(String id, boolean forceClaim)  throws TaskNotFoundException, InvalidStateException, InvalidOwnerException {
+        String userName = CurrentUserContext.getUserid();
+        return claim(id, userName, forceClaim);
+    }
+
+    public Task claim(String id, String userName, boolean forceClaim) throws TaskNotFoundException, InvalidStateException, InvalidOwnerException {
         LOGGER.debug("entry to claim(id = {}, userName = {})", id, userName);
         Task task = null;
         try {
             taskanaEngineImpl.openConnection();
             task = taskMapper.findById(id);
```

Comment:
Should call this.getTaskById(), because DRY-principle gets hurt, if we repeat exceptionhandling/reactions on other functionalities.